src/Console/AddCommand.php

```diff
@@ -22,25 +22,196 @@
+        $command->addOption('phone', null, InputOption::VALUE_OPTIONAL, 'Contact phone number');
+        $command->addOption('donationAmount', null, InputOption::VALUE_OPTIONAL, 'Monthly donation amount');
+        $command->addOption('comment', null, InputOption::VALUE_OPTIONAL, 'Comment');
     }
 
     public function execute(InputInterface $input, OutputInterface $output, ContainerInterface $container)
     {
-        // TODO implement...
+        $donorBuilder = $container->get('donor_builder');
+        $donorMapper = $container->get('donor_mapper');
+
+        $this->setPayerNumber(
+            $this->getProperty('payerNumber', 'Unique ID number for donor', $input, $output),
+            $donorBuilder
+        );
+        $this->setAccount(
+            $this->getProperty('account', 'Donor account number', $input, $output),
+            $donorBuilder
+        );
+        $this->setId(
+            $this->getProperty('id', 'Donor personal id number or organisation number', $input, $output),
+            $donorBuilder
+        );
+        $this->setName(
+            $this->getProperty('name', 'Donor name', $input, $output),
+            $donorBuilder
+        );
+        $this->setPostalAddress(
+            [
+                'address1' => $this->getProperty('address1', 'Donor Address line 1', $input, $output),
+                'address2' => $this->getProperty('address2', 'Donor Address line 2', $input, $output),
+                'postalCode' => $this->getProperty('postalCode', 'Donor Postal code', $input, $output),
+                'postalCity' => $this-> getProperty('postalCity', 'Donor Address city', $input, $output),
+                'coAddress' => $this->getProperty('coAddress', 'C/o Address', $input, $output),
+            ],
+            $donorBuilder
+        );
+        $this->setEmail(
+            $this->getProperty('email', 'Contact email address', $input, $output),
+            $donorBuilder
+        );
+        $this->setPhone(
+            $this->getProperty('phone', 'Contact phone number', $input, $output),
+            $donorBuilder
+        );
+        $this->setDonationAmount(
+            $this->getProperty('donationAmount', 'Monthly donation amount', $input, $output),
+            $donorBuilder
+        );
+        $this->setComment(
+            $this->getProperty('comment', 'Comment', $input, $output),
+            $donorBuilder
+        );
+
+        $donorMapper->save($donorBuilder->buildDonor());
+        $output->writeln('donor saved');
+    }
+
+    private function getProperty(
+        string $key,
+        string $desc,
+        InputInterface $input,
+        OutputInterface $output
+    ) {
+        $value = $input->getOption(str_replace('_', '-', $key));
+
+        if (!$value) {
+            $value = $this->command->getHelper('question')->ask(
+                $input,
+                $output,
+                new Question("$desc: ", '')
```

Comment:
The second argument to `new Question()` is the default value. That value is piced in interactive mode. So set a good default here and interactive mode is no problem..